src/app/space/create/deployments/apps/deployment-details.component.ts

```diff
@@ -143,13 +143,11 @@ export class DeploymentDetailsComponent {
   cpuChartClass: string;
   memChartClass: string;
 
-  private cpuChart: ChartAPI;
-  private memChart: ChartAPI;
-  private netChart: ChartAPI;
+  private cpuChart: Subject<ChartAPI> = new ReplaySubject<ChartAPI>();
+  private memChart: Subject<ChartAPI> = new ReplaySubject<ChartAPI>();
+  private netChart: Subject<ChartAPI> = new ReplaySubject<ChartAPI>();
 
-  private cpuChartLoad: Subject<void> = new ReplaySubject<void>();
-  private memChartLoad: Subject<void> = new ReplaySubject<void>();
-  private netChartLoad: Subject<void> = new ReplaySubject<void>();
+  private static NO_CHART: ChartAPI = {} as ChartAPI;
```

Comment:
I might actually prefer `null` over `{} as ChartAPI` as the latter can break the type assumption one has from function definitions. E.g.

```
  aFunction(a: boolean): X {
    if (a) { 
      return new X();
    }

    return {} as X;
  }
```

If X contains any non-optional properties, this function is very misleading to a user who doesn't read the implementation, which could involve many other inner functions. The more accurate definition would be
```
aFunction(a: boolean): X | {} {
  [...]
}
```

This increases the likelihood a user will expect and handle the empty object properly. I would try hard to not use `as` keyword outside of tests though.